react_app/src/reducers/auth.reducers.js

```diff
@@ -0,0 +1,56 @@
+import * as ActionTypes from "../actionTypes";
+
+const initialState = {
+  isAuthenticated: false,
+  userName: "",
+  isAdmin: false,
+  newUser: false,
+  errorMessage: null
+};
+
+/**
+ * action creators
+ */
+
+/**
+ * reducers
+ */
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case ActionTypes.CHECK_AUTHENTICATION_SUCCESS:
+      return {
+        ...state,
+        isAdmin: !!action.response.isAdmin,
+        isAuthenticated: !!action.response.isAuthenticated,
+        userName: action.response.userName,
+        errorMessage: null
+      };
+
+    case ActionTypes.LOGIN_SUCCESS:
+      return {
+        isAuthenticated: true,
+        userName: action.response.userName,
+        isAdmin: action.response.isAdmin
+      };
+
+    case ActionTypes.LOGIN_FAILED:
+      return {
+        ...initialState,
+        errorMessage: action.error
+      };
+
+    case ActionTypes.UPDATE_NEW_USER:
```

Comment:
I still think you should be using a router to handle "what's the user up to" type situations but I've just replicated your `newUser` state here for now with a reducer to update it.

Reply:
I don't see what adding a router really adds, but if you think it would be a good thing to do I'll have a look at doing it

Reply:
I'll pick that one up for you too if you want :) in terms of what it adds - you're perilously close to having all of your components render based on a piece of state / combination of state - you could drive all that through a router to keep it out of your state AND it means your users would be able to paste links to each other.